extensions/amp-image-viewer/0.1/amp-image-viewer.js

```diff
@@ -172,7 +172,8 @@ export class AmpImageViewer extends AMP.BaseElement {
       return Promise.resolve();
     }
     const ampImg = dev().assertElement(this.sourceAmpImage_);
-    const hasCompletedLayout = !!ampImg.querySelector('img');
+    const hasCompletedLayout = !!ampImg.querySelector(
```

Comment:
This is a very brittle test. Why not just call the `whenSignal`? If it's rendered, it'll be immediately resolved anyways.

Reply:
This check was originally added in https://github.com/ampproject/amphtml/pull/17145. I believe the reason why this check was added is because the `whenSignal` for `LOAD_END` never retriggers if `layoutCallback` for `<amp-img>` is called again after `unlayoutCallback`. What we actually want to wait for is for the `<img>` to be laid out on the page (i.e. for `layoutCallback` to be done).

Reply:
If the `amp-img` already laid out, it'll be [immediately resolved](https://github.com/ampproject/amphtml/blob/master/src/utils/signals.js#L71-L75). It's likely that bug is somewhere else.

Reply:
@sparhami and I discussed this a bit more offline (we went over the relevant portions of custom-elements.js and signals.js), and it seems like @jridgewell's proposed should work, and the test case to verify should be easy (https://github.com/ampproject/amphtml/issues/17044). I think the original regression was actually caused by code changes unrelated to lightbox, so maybe at some point in time the `whenSignal` for `LOAD_END` didn't trigger after `unlayoutCallback` but that behavior has changed since https://github.com/ampproject/amphtml/issues/17044 and #17145. We'll test and remove this constraint if the two test cases for reopening lightbox and lightbox with SSR still work.

Reply:
I changed to just use the signal, and verified re-opening the gallery, SSR and non-SRR work.